### PR TITLE
Refactor legacy monotonic time handling

### PR DESCRIPTION
**Pull request by robinkunde**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Replaces outdated/deprecated code for generating monotonic timestamp with the recommended approach. I also looked at the source code for the method to verify what it does https://opensource.apple.com/source/Libc/Libc-1158.1.2/gen/clock_gettime.c.auto.html.

Once we increase the minimum target to 10.12, we can switch over to CLOCK_MONOTONIC_RAW_APPROX, which is more accurate and faster because it doesn't require a system call.

### CODE REVIEW

**Comment by Jason-Morcos** [comment]
@robinkunde Are you planning on still working on this or should I close it?

**Comment by robinkunde** [comment]
Yes, I'm hoping to get back to it today.

**Review by Kaspik — APPROVED** [review]
LGTM, will leave it up to others for review too :)

**Comment by Jason-Morcos** [comment]
> Once we increase the minimum target to 10.12, we can switch over to CLOCK_MONOTONIC_RAW_APPROX, which is more accurate and faster because it doesn't require a system call.

How about we go ahead and this switch now and merge this into the main_3 branch for 3.0 release? Thoughts

**Review by Jason-Morcos — APPROVED** [review]
LGTM too! If it's working all good compiling locally, looks good to move forward

**Pull request merged**
